File: AIペアプログラミング実践編_ソースコード/practical-python/session-3/Ans/inventory/models/order.py
from django.db import models


class OrderStatus(models.TextChoices):
    """発注ステータス"""
    PENDING = 'PENDING', '発注中'
    ORDERED = 'ORDERED', '発注済み'
    RECEIVED = 'RECEIVED', '入荷済み'
    CANCELLED = 'CANCELLED', 'キャンセル'


class Order(models.Model):
    """発注情報"""
    product = models.ForeignKey(
        'Product',
        on_delete=models.CASCADE,
        verbose_name='商品'
    )
    quantity = models.PositiveIntegerField('発注数量')
    status = models.CharField(
        'ステータス',
        max_length=20,
        choices=OrderStatus.choices,
        default=OrderStatus.PENDING
    )
    ordered_at = models.DateTimeField('発注日時', null=True, blank=True)
    received_at = models.DateTimeField('入荷日時', null=True, blank=True)
    created_at = models.DateTimeField('作成日時', auto_now_add=True)

    class Meta:
        db_table = 'orders'
        verbose_name = '発注'
        verbose_name_plural = '発注'

    def __str__(self):
        return f'発注#{self.id} {self.product.name}'

File: AIペアプログラミング実践編_ソースコード/practical-ts/session-3/exercise/src/models/order.ts
/**
 * 発注ステータス
 */
export enum OrderStatus {
  /** 作成中 */
  PENDING = 'PENDING',
  /** 発注済み */
  ORDERED = 'ORDERED',
  /** 入荷済み */
  RECEIVED = 'RECEIVED',
  /** キャンセル */
  CANCELLED = 'CANCELLED',
}

/**
 * 発注情報
 */
export interface Order {
  /** 発注ID（自動採番） */
  id: number
  /** 商品ID */
  productId: number
  /** 発注数量 */
  quantity: number
  /** ステータス */
  status: OrderStatus
  /** 発注日 */
  orderedAt?: Date
  /** 入荷日 */
  receivedAt?: Date
  /** 作成日時 */
  createdAt: Date
}

/**
 * 発注作成時の入力
 */
export interface OrderInput {
  /** 商品ID */
  productId: number
  /** 発注数量 */
  quantity: number
}

/**
 * 発注更新時の入力（部分更新可能）
 */
export interface OrderUpdateInput {
  /** 発注数量 */
  quantity?: number
  /** ステータス */
  status?: OrderStatus
}
